### ISSUE TITLE
Warn when `trap action condition` is not correct

### ISSUE THREAD

**Issue by stdedos** [koalaman/shellcheck#2759]
#### For bugs
- Rule Id (if any, e.g. SC1000): 
- My shellcheck version (`shellcheck --version` or "online"): online
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
## Example: a typical script with several problems
trap rm -rf "$TMPFILE" EXIT
```

#### Here's what shellcheck currently says:

```console
$ shellcheck myscript
No issues detected!

$
```

#### Here's what I wanted or expected to see:

```console
$ shellcheck myscript
 
Line 3:
trap rm -rf "$TMPFILE" EXIT
        ^-- [SCxxxx](https://www.shellcheck.net/wiki/SCxxxx) (error): The `trap` action must be a singular command (akin to `eval`), _or a function_.

$
```

Somewhat related to https://github.com/koalaman/shellcheck/issues/2584, it'd be nice if `shellcheck` would remind me the syntax of traps.

Ofc I understand that detecting `condition` is hard.
I just wanted to put it out there.
